classes/Bulk/Bulk.php

```diff
@@ -598,15 +598,13 @@ public function bulk_get_stats_callback() {
 	 * @return void
 	 */
 	public function maybe_generate_missing_nextgen( $old_value, $value ) {
-		if ( isset( $old_value['convert_to_avif'] ) && isset( $value['convert_to_avif'] ) ) {
+		if ( empty( $old_value['convert_to_avif'] ) === empty( $value['convert_to_avif'] ) ) {
+			// Old value = new value so do nothing.
 			return;
 		}
 
-		if ( ! isset( $old_value['convert_to_avif'] ) && ! isset( $value['convert_to_avif'] ) ) {
-			return;
-		}
-
-		if ( isset( $value['convert_to_avif'] ) && ! (bool) $value['convert_to_avif'] ) {
+		if ( empty( $value['convert_to_avif'] ) ) {
+			// new value is disabled, do nothing.
 			return;
```

Comment:
aren't you missing the case where the old and new value are enabled, we should also bail-out?

Reply:
Isn't it catch by this?

```
if ( empty( $old_value['convert_to_avif'] ) === empty( $value['convert_to_avif'] ) ) {
			// Old value = new value so do nothing.
			return;
		}
```
		
Both empty evaluation would be false?

Reply:
if the value was previously enabled, and stayed enabled, both index will exist and not be empty

Reply:
Yes exactly, so:
empty( $old_value['convert_to_avif'] ) is False
empty( $value['convert_to_avif'] ) is False.
empty( $old_value['convert_to_avif'] ) === empty( $value['convert_to_avif'] ) is True then, no?

So we bail out. Is there a mistake here?

Reply:
ah right, morning fog I guess, it works

Reply:
☕  😄 
To be fair, this is not really straightforward to read 😬